File: Business Logic Layer(Backend)/Reservation/Reservation.Infra/Service/MenuItemReviewService.cs
using Reservation.Core.Data;
using Reservation.Core.Service;
using System;
using System.Collections.Generic;
using System.Text;

using Reservation.Core.Repository;
namespace Reservation.Infra.Service
{
    public class MenuItemReviewService : IMenuItemReviewService
    {
        private readonly IMenuItemReviewRepositroy _menuItemReviewRepositroy;
        public MenuItemReviewService(IMenuItemReviewRepositroy menuItemReviewRepositroy)
        {
            _menuItemReviewRepositroy = menuItemReviewRepositroy;
        }

        public bool DeleteMenuItemReview(decimal menuItemReviewId)
        {
            return _menuItemReviewRepositroy.DeleteMenuItemReview(menuItemReviewId);
        }

        public List<MenuItemReview> GetAllMenuItemReview()
        {
            return _menuItemReviewRepositroy.GetAllMenuItemReview();
        }

        public List<MenuItemReview> GetMenuItemReviewByMenuItemId(decimal menuItemId)
        {
            return _menuItemReviewRepositroy.GetMenuItemReviewByMenuItemId(menuItemId);
        }

        public List<MenuItemReview> GetMenuItemReviewByMenuItemReviewId(decimal menuItemReviewId)
        {
            return _menuItemReviewRepositroy.GetMenuItemReviewByMenuItemReviewId(menuItemReviewId);
        }

        public bool InsertMenuItemReview(MenuItemReview menuItemReview)
        {
            return _menuItemReviewRepositroy.InsertMenuItemReview(menuItemReview);
        }

        public bool UpdateMenuItemReview(MenuItemReview menuItemReview)
        {
            return _menuItemReviewRepositroy.UpdateMenuItemReview(menuItemReview);
        }
    }
}

File: Presentation Layer (frontend)/Reservation_angular/src/app/service/menuItemReview.service.ts

import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Router } from '@angular/router';
import { MenuItemReview } from "src/data/menuItemReview";
import { environment } from 'src/environments/environment';
@Injectable({
  providedIn: 'root'
})
export class MenuItemReviewService {

  headerDict={
    'Content-Type':'application/json',
    'Accept':'application/json'
    }
 requestOptions={
      headers:new HttpHeaders(this.headerDict)
      }
 constructor(public http:HttpClient,private route:Router) { }
InsertMenuItemReview( menuItemReview:MenuItemReview){
   console.log(menuItemReview.Menu_Item_Id)
   return this.http.post<boolean>(environment.apiUrl+"MenuItemReview/InsertMenuItemReview",menuItemReview,this.requestOptions);
}
UpdateMenuItemReview(menuItemReview:MenuItemReview ){
   return  this.http.put<boolean>(environment.apiUrl+"MenuItemReview/UpdateMenuItemReview",menuItemReview,this.requestOptions);

}
DeleteMenuItemReview(menuItemReviewId:number){
return this.http.delete<boolean>(environment.apiUrl+"MenuItemReview/DeleteMenuItemReview?menuItemReviewId="+menuItemReviewId,this.requestOptions);
}
GetAllMenuItemReview(){
  return  this.http.get<any[]>(environment.apiUrl+"MenuItemReview/GetAllMenuItemReview",this.requestOptions);
}
GetMenuItemReviewByMenuItemReviewId(menuItemReviewId:number){
      return  this.http.get<any[]>(environment.apiUrl+"MenuItemReview/GetMenuItemReviewByMenuItemReviewId?menuItemReviewId="+menuItemReviewId,this.requestOptions);

}
GetMenuItemReviewByMenuItemId(menuItemId:number){
   return this.http.get<any[]>(environment.apiUrl+"MenuItemReview/GetMenuItemReviewByMenuItemId?menuItemId="+menuItemId,this.requestOptions);

}

}
    
